Cargo.toml

```diff
@@ -27,8 +27,10 @@ crossbeam-channel = { version = "0.5.13", optional = true }
 http = { version = "0.2.9", optional = true }
 libc = { version = "0.2.158", optional = true }
 mio = { version = "1.0.2", features = ["net", "os-poll"], optional = true }
+native-tls = "0.2.12"
+rustls-pemfile = "2.1.2"
 httparse = { version = "1.8.0", optional = true }
 libaeron-sys = { version = "1.44.1", optional = true }
 rand = { version = "0.8.5", optional = true }
-tcp-stream = { version = "0.26.1", optional = true }
+tcp-stream = { version = "0.26.1", optional = true, features = ["native-tls"]  }
```

Comment:
We should make this a feature flag for this crate at some point so that users have a choice of the TLS impl being used.